### PR TITLE
Remove noImplicitAny in fast-foundation

### PR DESCRIPTION
**Pull request by m4thieulavoie**
# Pull Request

## 📖 Description

This PR re-enables `noImplicitAny` in `fast-foundation`. To avoid typescript from being too nitpicky, I disabled `suppressImplicitAnyIndexErrors`.

### 🎫 Issues

https://github.com/microsoft/fast/issues/5338

## 👩‍💻 Reviewer Notes

I purposely set some `unknown`s in some `*Changed()` functions. I can change that if necessary 👍 

### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-foundation/src/accordion/accordion.ts [hunk 1]
```diff
@@ -54,14 +54,14 @@ export class Accordion extends FoundationElement {
...
-    public accordionItemsChanged(oldValue, newValue): void {
+    public accordionItemsChanged(oldValue: HTMLElement[], newValue: HTMLElement[]): void {
         if (this.$fastController.isConnected) {
             this.removeItemListeners(oldValue);
             this.setItems();
         }
     }
 
-    private activeid: string | undefined;
+    private activeid: string | null;
```
[the remark this answers is not in the document]
Comment (m4thieulavoie) [reply packages/web-components/fast-foundation/src/accordion/accordion.ts:64]: I'll jump back into this one either tonight or tomorrow and will look into it, I can't recall on top of my head 👍

[the remark this answers is not in the document]
Comment (m4thieulavoie) [reply packages/web-components/fast-foundation/src/accordion/accordion.ts:64]: So after digging in a little bit : it _is_ accurate (we assign the result of an `.getAttribute` to it at some point).

**Review by chrisdholt — APPROVED** [review]

**Comment by m4thieulavoie** [comment]
I just need to fix the CI and we'll be good to ship this one!

**Comment by EisenbergEffect** [comment]
Thanks @m4thieulavoie !

**Pull request merged**